src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -619,13 +607,13 @@ protected void clipForOverlappingReads(final SAMRecord read1, final SAMRecord re
                     if (posDiff > 0) {
                         CigarUtil.softClip3PrimeEndOfRead(pos, Math.min(pos.getReadLength(),
                                 pos.getReadLength() - posDiff + 1));
-                        calculateMdAndNmTags(pos);
+                        removeNmMdAndUqTags(pos); // these tags are no invalid!
```

Comment:
I think this a lot of effort for little reward, since likely one would need to run `SetNmMdAndUqTags` anyhow.